### ISSUE TITLE
IGV issue with colouring on TLEN / insert size & orientation

### ISSUE THREAD

**Issue by dnil** [Clinical-Genomics/scout#4548]
Somethings not quite stable with the color patterns for insert size deviation. One of the ends get the color, the other not from start. And it changes when selecting `View mate`. Perhaps a missing abs value somewhere. It might just be fixed on the current IGV, and we really haven't updated in a while anyway - lots of nice new features available!

See e.g. this [variant](https://scout.scilifelab.se/cust002/F0047445-4/sv/variants/81b4bcc1d464f6acf29160357a8379d6).
![Screenshot 2024-04-04 at 14 39 08](https://github.com/Clinical-Genomics/scout/assets/758570/9345badb-3e26-4f3d-87c3-9c9032e3a990)

![Screenshot 2024-04-04 at 14 39 20](https://github.com/Clinical-Genomics/scout/assets/758570/b7b43a64-1c01-49ee-a2a4-941692474695)

**Comment by dnil** [comment]
Hm, not easy to recreate based on checking out a few DEL events on stage. Situational then, perhaps based on a combo of supplemental alignment and pairing?

**Comment by dnil** [comment]
Replicated by loading the same case on stage. The latest release of IGV does not fix the issue.

**Comment by northwestwitch** [comment]
Have you asked on igv.js about this?

**Comment by dnil** [comment]
No, doubt there is much use until we can reproduce consistently? I only have one case so far.

**Comment by dnil** [comment]
A second example has been seen - it must be a bug. 
https://scout.scilifelab.se/cust002/F0060974/sv/variants/b59969b317b68a96ef902f5960b2e26f

**Comment by dnil** [comment]
Ok, we have a third. I'll see if I can't distill it into a bug report. https://scout38.sys.scilifelab.se/cust002/F0082538/sv/variants/f39c99fb03749f6db54116331f88c919

So, I fired up a local build of the current igv.js master. It does not show the issue with the same cram file. BUT that is with local serving of said file. It is quite possible there is some unrealised promise/range header issue happening on the live server. It is however a few changes ahead of our 3.8.7, so I guess I will try the 3.8.8 on stage as well for good measure.

**Comment by dnil** [comment]
Tested with a locally built `igv.js.min`. When adding debug output to console around https://github.com/igvteam/igv.js/blob/0fe2e26c0ba3414057ddcb021e527db28db57690/js/bam/alignmentTrack.js#L1454 the issue goes away. Tentatively guessing the `alignment.mate` somehow not always being fully fetched from its cram slice at this point to be the problem. Testing with a mildly modified version not relying on `mate` to be set for the insert size check.
